packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -9,14 +9,21 @@
 /* eslint-disable no-useless-escape */
 const crypto = require('crypto');
 const _ = require('lodash');
-const { sanitizeEntity } = require('@strapi/utils');
+const { sanitize } = require('@strapi/utils');
 const { getService } = require('../utils');
 
 const emailRegExp = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
 const formatError = error => [
   { messages: [{ id: error.id, message: error.message, field: error.field }] },
 ];
 
+const sanitizeUser = (user, ctx) => {
+  const { auth } = ctx.state;
+  const userSchema = strapi.getModel('plugin::users-permissions.user');
+
+  return sanitize.contentAPI.output(user, userSchema, { auth });
```

Comment:
I find it strange to have called that function contentAPI.output and using it in a lot of places that are really the content api 🤔

Reply:
It depends if we're including extensions of the content API in its definition. From my perspective, we're adding auth controllers from the UP plugin for the content API, thus we sanitize its result using contentAPI.output.
I can still create a sanitize dedicated to users-permissions controller next to them though, what do you think about it?

Reply:
I just don't think this is really realted to the contentAPI but more so the entities.